### PR TITLE
My combined fixes for a few issues

### LINKED ISSUE #9: [feature request] Print out URL of created PRs
It would be handy if the command line can print out the URL of created PRs, so that I can open the PR from command line directly rather than having to open the PR list to find or checking the email.

### LINKED ISSUE #17: Suggestion: ToC table for readability
||PR|Description|
|--|--|--|
||#1|Foo|
|👉|#17|Bar|
||#3|Baz|
||#4|Qux|
||#5|Quux|

### LINKED ISSUE #19: Bug: TypeError: Cannot read property 'match' of null
Sometimes I would get the error message `TypeError: Cannot read property 'match' of null`. e.g.:

```
All changes pushed ✅

This will create (or update) PRs for the following branches:
  -> annie/reset-pw (x1)
  -> annie/reset-pw-2_5 (x2)
  -> annie/reset-pw3 (x3)

Shall we do this? [y/n]  y

Checking if PR for branch annie/reset-pw already exists... yep
Checking if PR for branch annie/reset-pw-2_5 already exists... yep
Checking if PR for branch annie/reset-pw3 already exists... yep
Updating PR for branch annie/reset-pw...✅
❌  An error occured. Was there a conflict perhaps?
error TypeError: Cannot read property 'match' of null
    at upsertNavigationInBody (/usr/local/lib/node_modules/git-pr-train/github.js:70:12)
    at allBranches.reduce (/usr/local/lib/node_modules/git-pr-train/github.js:200:21)
    at process._tickCallback (internal/process/next_tick.js:68:7)
```

I believe the issue is that `memo.body` can be null.

https://github.com/realyze/pr-train/blob/8fdaa9a27e6e469e42b2d4897db2b0a90308ef69/github.js#L176

### LINKED ISSUE #24: Allow configuring a different name for the master branch
Some projects use a different naming convention for the `master` branch. Currently, pr-train cannot be used in those projects. Therefore, allow configuring the `master` branch name.

### LINKED ISSUE #25: [feature request] Support creation of draft PRs
If a user's plan supports it, the tool should allow PRs to be created in draft mode so they can be finalized and additional details added before everyone on the team gets notified about them.

## MVP

A new command-line flag to create PRs in draft mode, `--draft, -d`.

## Nice to have

It would also be good to be able to specify that PRs should be created in draft mode by default in the config file. That would save people from having to always specify the command-line flag.

### PR DESCRIPTION
**Pull request by greglockwood**
<pr-train-toc>

|     | PR | Description                                          |
| --- | -- | ---------------------------------------------------- |
|     | #30 | Added ability to customise the base branch          |
|     | greglockwood#3 | Draft PR support                                     |
|     | greglockwood#4 | Cater for body being nullish or empty                |
|     | greglockwood#5 | Putting the table back in 'table of contents'!       |
|     | greglockwood#6 | Added support to print links to PRs in terminal      |
|     | greglockwood#7 | Work when remote URL doesn't have .git suffix        |
| 👉  | #31 | **[combined branch]**                               |

</pr-train-toc>

Fixes #9 
Fixes #17
Fixes #19
Fixes #24
Fixes #25

Combines all changes from individual PRs in the train shown above.

### CODE REVIEW

**Comment by greglockwood** [comment]
@realyze I updated this over the weekend with all of your suggested changes on my PRs.

I don't know what your versioning strategy is with changes like these.

If you're happy with the result, feel free to merge it and bump the version as you see fit.

**Comment by realyze** [comment]
Hi @greglockwood , sorry for the delay, lots of things came up; I will have another look this week! Thanks again.

File: github.js [hunk 1]
```diff
@@ -67,21 +71,47 @@ function readGHKey() {
...
 }
 
+function checkAndReportInvalidBaseError(e, base) {
+  const { field, code } = get(e, 'body.errors[0]', {});
+  if (field === 'base' && code === 'invalid') {
+    console.log([
+      emoji.get('no_entry'),
+      `\n${emoji.get('confounded')} This is embarrassing. `,
+      `The base branch of ${base.bold} doesn't seem to exist on the remote.`,
+      `\nDid you forget to ${emoji.get('arrow_up')}  push it?`,
```
Comment (realyze) [inline github.js:89]:
```suggestion
      `\nDid you forget to ${emoji.get('arrow_up')} push it?`,
```

**Comment by realyze** [comment]
Thanks a lot for the PRs @greglockwood , I merged in your change and published as `git-pr-train@2.5.1` on `npm`. 🙇

**Pull request merged**